src/detect.c

```diff
@@ -984,6 +984,7 @@ static DetectRunScratchpad DetectRunSetup(
             if (p->proto == IPPROTO_TCP && pflow->protoctx &&
                     StreamReassembleRawHasDataReady(pflow->protoctx, p)) {
                 p->flags |= PKT_DETECT_HAS_STREAMDATA;
+                flow_flags |= STREAM_FLUSH;
```

Comment:
Could we have some setup (small stream depth) where the stream_flush will cause the inspection to be done in the middle of a header ? Or are we sure we have post header data at that point ?

Reply:
Headers are not affected. Body data is reinspected until 'min_inspect_size' is reached.